zerver/actions/realm_settings.py

```diff
@@ -56,6 +56,9 @@ def do_set_realm_property(
     ), f"Cannot update {name}: {value} is not an instance of {property_type}"
 
     old_value = getattr(realm, name)
+    if old_value == value:
+        return
```

Comment:
Our usual pattern with `do_` methods is to make the caller responsible for checking -- would it make sense to do an assert here instead?

Reply:
Ah, fair, `do_set_push_notifications_enabled_end_timestamp` also does the `if old_timestamp == value: return` thing - should it follow the `assert` convention? And in this block in ``
```
        for realm in Realm.objects.filter(push_notifications_enabled=True):
            do_set_realm_property(realm, "push_notifications_enabled", False, acting_user=None)
            do_set_push_notifications_enabled_end_timestamp(realm, None, acting_user=None)
```

we should be checking  if these need to be called at all?

Reply:
I think that would probably be better; but it may be worth a #backend thread.

I'll go ahead and merge this version; I think it's harmless from a production standpoint and just a question of how we want to help developers not make mistakes.